agent/consul/server_connect.go

```diff
@@ -104,6 +103,13 @@ func (s *Server) getCARoots(ws memdb.WatchSet, state *state.Store) (*structs.Ind
 	// data such as key material stored within the struct. So here we
 	// pull out some of the fields and copy them into
 	for i, r := range indexedRoots.Roots {
+		var intermediates []string
+		if r.IntermediateCerts != nil {
+			intermediates = make([]string, 0)
```

Comment:
minor: we know the eventual length, so it doesn't hurt to preallocate
```suggestion
			intermediates = make([]string, 0, len(r.IntermediateCerts))
```